src/wowman/config.clj

```diff
@@ -0,0 +1,80 @@
+(ns wowman.config
+  (:require
+   [me.raynes.fs :as fs]
+   [clojure.spec.alpha :as s]
+   [orchestra.spec.test :as st]
+   [orchestra.core :refer [defn-spec]]
+   [taoensso.timbre :as timbre :refer [debug info warn error spy]]
+   [spec-tools.core :as spec-tools]
+   [wowman
+    [specs :as sp]
+    [utils :as utils]]))
+
+(def default-cfg
+  {:addon-dir-list []
+   :debug? false
+   :selected-catalog :short})
+
+(defn handle-install-dir
+  "when `:install-dir` is present in given configuration, expands it to an addon-dir map and removes the value from the config
+  `:install-dir` was once supported in the user configuration but is now only supported in the command line options.
+  this function handles it's presence in both"
+  [cfg]
+  (let [install-dir (:install-dir cfg)
+        addon-dir-list (->> cfg :addon-dir-list (map :addon-dir) vec)
+        stub {:addon-dir install-dir :game-track "retail"}
+        ;; add stub to addon-dir-list if install-dir isn't nil and doesn't match anything already present
+        cfg (if (and install-dir
+                     (not (utils/in? install-dir addon-dir-list)))
+              (update-in cfg [:addon-dir-list] conj stub)
+              cfg)]
+      ;; finally, ensure :install-dir is absent from whatever we return
+    (dissoc cfg :install-dir)))
+
+(defn remove-non-existant-dirs
+  "removes any addon directories from the given configuration that do not exist"
```

Comment:
removes any `addon-dir-map` items from the given configuration whose directories do not exist